shared/video-sdk/develop/audio-and-voice-effects/project-implementation/unity.mdx

```diff
@@ -0,0 +1,251 @@
+    private string audioFilePath = "https://www.kozco.com/tech/organfinale.mp3"; // URL or path to the audio mixing file
+    private TMP_Text audioMixing, playEffectButton, voiceEffectButton, speakerphoneSwitch;
+    private bool isChecked = false;
+    ```
+
+1. **Set up access to the UI elements**
+
+    In `Assets/Agora-RTC-Plugin/Agora-Unity-RTC-SDK/Code/JoinChannelVideo.cs`, add the following at the end of `SetupUI();`
+
+    ```csharp
+    // Setup access to the audio mixing button.
+    go = GameObject.Find("audioMixing");
+    go.GetComponent<Button>().onClick.AddListener(AudioMixing);
+    audioMixing = go.GetComponentInChildren<TextMeshProUGUI>(true);
+    audioMixing.text = "Play Audio";
+    audioMixing.fontSize = 14;
+    // Setup access to the voice effects button.
+    go = GameObject.Find("voiceEffects");
+    go.GetComponent<Button>().onClick.AddListener(playSoundEffect);
+    playEffectButton = go.GetComponentInChildren<TextMeshProUGUI>(true);
+    playEffectButton.text = "Play sound effects";
+    playEffectButton.fontSize = 14;
+    // Setup access to the audio effects button.
+    go = GameObject.Find("audioEffects");
+    go.GetComponent<Button>().onClick.AddListener(applyVoiceEffect);
+    voiceEffectButton = go.GetComponentInChildren<TextMeshProUGUI>(true);
+    voiceEffectButton.text = "Play audio effects";
+    voiceEffectButton.fontSize = 14;
+    // Toggle switch to enable or disable the speaker.
+    go = GameObject.Find("playbackDevice");
+    go.GetComponent<Toggle>().isOn = true;
+    go.GetComponent<Toggle>().onValueChanged.AddListener((value) =>
+    {
+        playbackSpeaker(value);
+    });
+    ```
+
+### Implement sound and voice effects
+
+To add sound and voice effect to your <Vpl k="CLIENT" />, take the following steps:
+
+1. **Enable the user to start and stop audio mixing**
+        
+    Add the following method to the `JoinChannelVideo` class:
+        
+    ```csharp
+    public void AudioMixing() {  
+        audioPlaying = !audioPlaying;
+
+        if (audioPlaying) {
+            audioMixing.text = "Stop playing audio";
+            RtcEngine.StartAudioMixing(audioFilePath, false, false, -1);
+            Debug.Log("Audio playing");
+        } else {
+            RtcEngine.StopAudioMixing();
+            audioMixing.text = "Play Audio";
+        }
+    }
+    ```
+
+1. **Pre-load sound effects**
+
+    To set up playing voice effects, pre-load the sound effects. To add this, in `JoinChannelVideo`, add the following lines at the end of the `setupVideoSDKEngine` method:
+
+    ```csharp
+    // Pre-load sound effects to improve performance
+    RtcEngine.PreloadEffect(soundEffectId, soundEffectFilePath);
+    ```
+
+1. **Play, pause, or resume playing the sound effect**
+
+    When a user presses the button, the sound effect is played. When they press the button again, the effect is paused and resumed alternately. To do this, add the following method to the `JoinChannelVideo` class:
```

Comment:
To implement this workflow, add ... 


Can you update all of the "To do this" for something else please.